docs/CONTRIBUTING.md

```diff
@@ -0,0 +1,90 @@
+- [Set up Git](https://docs.github.com/en/get-started/quickstart/set-up-git)
+- [GitHub flow](https://docs.github.com/en/get-started/quickstart/github-flow)
+- [Collaborating with pull requests](https://docs.github.com/en/github/collaborating-with-pull-requests)
+- [How to Write GitHub Markdown](contributing/content-markup-reference.md).
+
+## Getting started
+
+### Issues
+
+#### Create a new issue
+
+If you spot a problem with the docs, [search if an issue already exists](https://docs.github.com/en/github/searching-for-information-on-github/searching-on-github/searching-issues-and-pull-requests#search-by-the-title-body-or-comments). If a related issue doesn't exist, you can open a new issue using a relevant [issue form](https://github.com/cuttle-cards/cuttle/issues/new/choose).
+
+#### Solve an issue
+
+Scan through our [existing issues](https://github.com/cuttle-cards/cuttle/issues) to find one that interests you. You can narrow down the search using `labels` as filters. See [Labels](/contributing/how-to-use-labels.md) for more information. As a general rule, we don’t assign issues to anyone. If you find an issue to work on, you are welcome to open a PR with a fix.
+
+### Make Changes
+
+#### Make changes directly on Github
+
+Click the **PENCIL** icon top right of any docs page to make small changes such as a typo, sentence fix, or a broken link. This takes you a file editor where you can make your changes and [create a pull request](#pull-request) for a review.
+
+#### Make changes locally
+
+1. [Install Git LFS](https://docs.github.com/en/github/managing-large-files/versioning-large-files/installing-git-large-file-storage).
+
+2. Fork the repository.
+
+- Using the command line:
+  - [Fork the repo](https://docs.github.com/en/github/getting-started-with-github/fork-a-repo#fork-an-example-repository) so that you can make your changes without affecting the original project until you're ready to merge them.
+
+3. Install or update to **[Node.js v16](https://nodejs.org/en/download/)**.
+
+   At the root directory, use the cmd
+
+   ```bash
+   npm ci
+   ```
+
+4. Set up Cypress.io, our end-to-end testing framework, please follow the [installation guide](https://docs.cypress.io/guides/getting-started/installing-cypress#System-requirements) according to your operating system.
+
+5. For seeding the database, here's a quick guide:
+
+   While booted in development mode (the default), you can make requests to `/test/loadSeasonFixture` and `/test/loadMatchFixtures` to seed records into those tables.
+
+   The easiest way to do it is using test code. If you look at `tests/e2e/specs/out-of-game/stats.spec.js` you'll find the tests for the stats page, which are written using [cypress](https://docs.cypress.io). These tests already import fixture data from `tests/e2e/fixtures/statsFixtures.js` and use a custom cypress command to request those endpoints and seed that data into the database `beforeEach` test.
```

Comment:
The format of my suggestion got fucked up b/c I was trying to embed code snippets in it, but you get the idea

Reply:
Yeah, it was definitely easier to spin up the tests and make changes, since Cypress "plays" the game for you.

I think we can move the testing up higher in the instructions list. Maybe I can state that this project prefers the BTD over direct development?

Reply:
I think moving testing up makes sense. I usually consider it test-driven development, but yeah I think that sounds good. Maybe something like

> We encourage contributors to practice Test Driven Development (TDD) when working on cuttle.cards, because it can save an enormous amount of time while increasing everyone's confidence that the application does what it's supposed to. By utilizing the existing testing infrastructure, you can quickly and automatically do things like:

1. Signup two accounts and drop them into a game
2. Load the game into a specific state i.e. putting the cards wherever you want
3. Make arbitrary chains of moves on behalf of both players
4. Assert the full state of the game (where the relevant cards are)
and more. 

The general idea behind Test Driven Development is to write/modify the tests before making changes to the application. This lets you clearly define what the application is supposed to do before changing things, so that you'll know with confidence when things work correctly. TDD takes practice, and it's not required to contribute, but we are happy to help you practice it and once you get going with testing, you'll see what an enormous QoL improvement it can make to your developer experience

Reply:
Okay, I'll make that change shortly.

Reply:
I've moved this recent section you wrote above, and have the instructions on how to perform that task below the set up instructions.